src/foundations/ui/slider/slider.tsx

```diff
@@ -0,0 +1,201 @@
+"use client";
+
+import { cva, cn } from "@/lib/utils";
+
+interface SliderProps
+  extends Omit<React.ComponentPropsWithRef<"input">, "onChange" | "disabled"> {
+  min?: number;
+  max?: number;
+  step?: number;
+  value: number;
+  onChange?: (value: number) => void;
+  orientation?: "horizontal" | "vertical";
+  disabled?: boolean;
+}
+
+const containerStyle = cva({
+  base: [
+    // CSS variables
+    "[--track-thickness:--spacing(2)]",
+    "[--thumb-size:--spacing(4)]",
+
+    "relative flex",
+  ],
+  variants: {
+    variant: {
+      horizontal: "items-center w-full h-[var(--track-thickness)]",
+      vertical: "justify-center w-[var(--track-thickness)] h-full",
+    },
+  },
+  defaultVariants: {
+    variant: "horizontal",
+  },
+});
+
+const inputStyle = cva({
+  base: [
+    "absolute cursor-pointer appearance-none rounded-full",
+
+    // thumb
+    // webkit (chrome, safari)
+    "[&::-webkit-slider-thumb]:appearance-none",
```

Comment:
```suggestion
    "[&::-webkit-slider-thumb]:appearance-none",
    "[&::-webkit-slider-thumb]:transition-transform",
```

Simple transitions for the pseudo-states would go a long way